src/index.d.ts

```diff
@@ -0,0 +1,825 @@
+// Type definitions for node-libtidy v0.3
+// Project: node-libtidy
+// Definitions by: Wang Guan <momocraft@gmail.com> / https://github.com/jokester
+
+export const tidyBuffer: TidyBufferStatic
+export const TidyDoc: TidyDocConstructor
+export const TidyOption: TidyOptionConstructor
+export const compat: TidyCompat
+/// <reference types="node" />
+
+/**
+ *  Callback convention: the result
+ */
+interface TidyResult {
+    /**
+     * errlog contains the error messages generated during the run,
+     * formatted as a string including a trailing newline.
+     */
+    errlog?: string
+    /**
+     * output contains the output buffer if output was generated.
+     * The property is unset if generating output was not part of the method
+     * in question, or null if no output was generated due to errors.
+     */
+    output?: Buffer
+}
+
+/**
+ * Callback convention: the signerature used in async APIs
+ */
+interface TidyCallback {
+    (err: Error | null, res: TidyResult | null): void
+}
+
+/**
+ * High-level functions automate the most common workflows.
+ *
+ * The document is assumed to be a buffer or a string.
+ * Anything else will be converted to a string and then
+ * turned into a buffer.
+ */
+interface TidyBufferStatic {
+    (document: string | Buffer, options: Generated.OptionDict,
+        callback: TidyCallback): void
+
+    (document: string | Buffer, callback: TidyCallback): void
+}
+
+interface TidyOption {
+    readonly name: string
+    readonly type: "boolean" | "integer" | "doctype" | "string"
+    readonly readOnly: boolean
+    readonly pickList: string[]
+    readonly id: number
+    toString(): string
+}
+
+/**
+ * TidyOption type and constructor (not for public use)
+ */
+interface TidyOptionConstructor {
+    new (): TidyOption
+    (): TidyOption
+}
+
+/**
+ * TidyDoc is the central object for dealing with the library at a low level
+ *
+ * Such an object will hold a configuration and be able to process one input
+ * file at a time, while multiple such objects can deal with multiple inputs
+ * simultaneously using an independent configuration for each of them.
+ */
+interface TidyDoc extends Generated.TidyDocOption {
+    // Sync calls
+    cleanAndRepairSync(): string
+    parseBufferSync(document: Buffer): string
+    runDiagnosticsSync(): string
+    saveBufferSync(): Buffer
+    getErrorLog(): string
```

Comment:
The `getErrorLog` method is currently not documented, because it should not be needed for normal operations. Every error should either be returned from a sync call or passed to the callback of an async call. It's just for the sake of a unit test that I allow access to this, to ensure we're not generating additional errors that we fail to report. The method may disappear at any moment. Perhaps I should rename it to `_getErrorLog` to communicate that idea.